### PR TITLE
Simplify alica libraries

### PR DESCRIPTION
**Pull request by dhananjay-patki**
Remove Fsystem and alica_common_config libraries

### CODE REVIEW

File: alica_dummy_proxy/CMakeLists.txt [hunk 1]
```diff
@@ -10,17 +10,12 @@ set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}/cmake)
 
 
 # Imports
-find_package(alica_common_config REQUIRED)
```
Comment (Dale-Koenig) [inline alica_dummy_proxy/CMakeLists.txt:13]: I think .editorconfig can just be removed (Comment in the wrong place because I can't comment on deleted file)

File: alica_engine/src/engine/RunningPlan.cpp [hunk 2]
```diff
@@ -339,7 +337,6 @@ void RunningPlan::useEntryPoint(const EntryPoint* value)
 void RunningPlan::useState(const State* s)
 {
     if (_activeTriple.state != s) {
-        ALICA_ASSERT(s == nullptr || (_activeTriple.entryPoint && _activeTriple.entryPoint->isStateReachable(s)));
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/RunningPlan.cpp:341]: Can maybe preserve as a normal `assert`

File: alica_common_config/CMakeLists.txt [hunk 3]
```diff
@@ -1,41 +0,0 @@
-cmake_minimum_required(VERSION 3.5.1)
-project(alica_common_config)
-
-add_compile_options(-std=c++14)
-
-# Set up utils to allow Colcon/catkin to find the packages
-include(GNUInstallDirs)
-include(CMakePackageConfigHelpers)
-set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}/cmake)
```
Comment (Dale-Koenig) [inline alica_common_config/CMakeLists.txt:9]: How is the cflags file that was in this package being used now?

Comment (dhananjay-patki) [reply alica_common_config/CMakeLists.txt:9]: Was migrated to the solver interface package

File: alica_engine/src/engine/RunningPlan.cpp (hunk 2 above)
Comment (dhananjay-patki) [reply alica_engine/src/engine/RunningPlan.cpp:341]: I don't think that's what what the original macro did, though.
ALICA_ASSERT(CHECK) (LIKELY(CHECK) ? void(0) : [] { assert(!#CHECK); }()) is the definition - if the check isn't true, then it asserts !check, so as far as I can tell the macro never really did anything?

Comment (Dale-Koenig) [reply alica_engine/src/engine/RunningPlan.cpp:341]: That means logic wise it is the same as a normal assert right?

File: alica_common_config/CMakeLists.txt (hunk 3 above)
Comment (Dale-Koenig) [reply alica_common_config/CMakeLists.txt:9]: I'm more wondering what causes it to actually be used

File: alica_engine/src/engine/RunningPlan.cpp (hunk 2 above)
Comment (dhananjay-patki) [reply alica_engine/src/engine/RunningPlan.cpp:341]: Logic wise I think I think its like 
if cond { void(0) } (no-op)
else { assert(!cond) }
but to make it in to the else block the condition has to be false, so the assert is pointless.

If we replace it with a regular assert, then if the condition is false, it would terminate, but the macro would just assert (!false) and do nothing

File: alica_common_config/CMakeLists.txt (hunk 3 above)
Comment (dhananjay-patki) [reply alica_common_config/CMakeLists.txt:9]: It should be detected by the cmake config helpers, but it looks like it's only working for this package and not for all of them. Looking at a fix.

Comment (dhananjay-patki) [reply alica_common_config/CMakeLists.txt:9]: Looks like there was a bug in how we were doing it previously - not all the packages had the compilation flags enabled. That's been fixed by turning the cflags into a macro and calling it in all the packages

**Review by Dale-Koenig — APPROVED** [review]

File: cmake_flags/cflags.cmake [hunk 4]
```diff
@@ -1,10 +1,11 @@
+macro(alica_compile_flags)
```
Comment (triccyx) [inline cmake_flags/cflags.cmake:1]: I would prefer to create a root CMakeList.txt with these configurations. Is this not possible?

File: alica_tests/CMakeLists.txt [hunk 5]
```diff
@@ -4,35 +4,34 @@ project(alica_tests)
 add_compile_options(-std=c++17)
 set(CMAKE_BUILD_TYPE Debug)
 
+include(../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline alica_tests/CMakeLists.txt]: See comment here
cmake_flags/cflags.cmake

File: supplementary/supplementary_tests/CMakeLists.txt [hunk 6]
```diff
@@ -3,14 +3,17 @@ project(supplementary_tests)
 
 add_compile_options(-std=c++17)
 
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/supplementary_tests/CMakeLists.txt:6]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/constraintsolver/CMakeLists.txt [hunk 7]
```diff
@@ -3,6 +3,9 @@ project(constraintsolver)
 
 add_compile_options(-std=c++17)
 
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/constraintsolver/CMakeLists.txt:6]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/autodiff/CMakeLists.txt [hunk 8]
```diff
@@ -7,6 +7,8 @@ if (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
 	add_definitions (-DCMAKE_ECLIPSE_GENERATE_SOURCE_PROJECT=TRUE)
 endif (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
 
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/autodiff/CMakeLists.txt:10]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/alica_tracing/CMakeLists.txt [hunk 9]
```diff
@@ -2,10 +2,10 @@ cmake_minimum_required(VERSION 3.5.1)
 project(alica_tracing)
 add_compile_options(-std=c++17)
 
-set(PKG_DEPS
-)
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/alica_tracing/CMakeLists.txt:5]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/alica_ros_turtlesim/CMakeLists.txt [hunk 10]
```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.5.1)
 project(alica_ros_turtlesim)
 add_compile_options(-std=c++17)
 
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/alica_ros_turtlesim/CMakeLists.txt:5]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/alica_dummy_tracing/CMakeLists.txt [hunk 11]
```diff
@@ -2,7 +2,8 @@ cmake_minimum_required(VERSION 3.10)
 project(alica_dummy_tracing)
 
 add_compile_options(-std=c++17)
-
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/alica_dummy_tracing/CMakeLists.txt:5]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/alica_ros_proxy/CMakeLists.txt [hunk 12]
```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.5.1)
 project(alica_ros_proxy)
 
 add_compile_options(-std=c++17)
+include(../../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline supplementary/alica_ros_proxy/CMakeLists.txt:5]: See comment here:
cmake_flags/cflags.cmake

File: supplementary/alica_ros_proxy/CMakeLists.txt [hunk 13]
```diff
@@ -26,7 +27,6 @@ catkin_package(
 include_directories(
 	include
 	${catkin_INCLUDE_DIRS}
-  ${alica_common_config_INCLUDE_DIRS}
   ${alica_solver_interface_INCLUDE_DIRS}
```
Comment (triccyx) [inline supplementary/alica_ros_proxy/CMakeLists.txt:30]: I think you can remove for find_package (alica_engine)

File: alica_simple_solver/CMakeLists.txt [hunk 14]
```diff
@@ -9,6 +9,8 @@ if (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
 	add_definitions (-DCMAKE_ECLIPSE_GENERATE_SOURCE_PROJECT=TRUE)
 endif (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
 
+include(../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline alica_simple_solver/CMakeLists.txt:12]: See comment here
cmake_flags/cflags.cmake

File: alica_solver_interface/CMakeLists.txt [hunk 15]
```diff
@@ -15,18 +15,14 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}/cmake) 
 
-
-# Imports 
-find_package(alica_common_config REQUIRED)
-
+include(${CMAKE_CURRENT_SOURCE_DIR}/../cmake_flags/cflags.cmake)
```
Comment (triccyx) [inline alica_solver_interface/CMakeLists.txt:18]: See comment here
cmake_flags/cflags.cmake

File: alica_solver_interface/CMakeLists.txt [hunk 16]
```diff
@@ -74,6 +70,8 @@ install(DIRECTORY include/${PROJECT_NAME}/
 )
 
 
+message("${CMAKE_CXX_FLAGS}")
```
Comment (triccyx) [inline alica_solver_interface/CMakeLists.txt]: Can be removed

File: supplementary/constraintsolver/CMakeLists.txt [hunk 17]
```diff
@@ -21,13 +22,12 @@ catkin_package(
...
  CATKIN_DEPENDS   roslib 
- DEPENDS fsystem alica_common_config alica_solver_interface autodiff alica_engine
+ DEPENDS  alica_solver_interface autodiff alica_engine
 )
 
 include_directories(
   include
   ${catkin_INCLUDE_DIRS}
-  ${alica_common_config_INCLUDE_DIRS}
   ${alica_engine_INCLUDE_DIRS}
```
Comment (triccyx) [inline supplementary/constraintsolver/CMakeLists.txt]: Can be removed for find_package

**Comment by triccyx** [comment]
I noticed in https://github.com/rapyuta-robotics/alica/blob/0b0fb6269f0073655b96f269f0ddc81cc1d22595/supplementary/alica_tracing/CMakeLists.txt#L25
That alica_engine can be removed due to find_package
Also here:
https://github.com/rapyuta-robotics/alica/blob/0b0fb6269f0073655b96f269f0ddc81cc1d22595/supplementary/constraintsolver/CMakeLists.txt#L31

File: cmake_flags/cflags.cmake (hunk 4 above)
Comment (dhananjay-patki) [reply cmake_flags/cflags.cmake:1]: It's technically possible, but in that case we have to bundle the entire alica repo as 1 package. If we want to be able to build ros1 and ros2 builds from the same branch, I can see that causing problems down the road.

File: supplementary/alica_ros_proxy/CMakeLists.txt (hunk 13 above)
Comment (dhananjay-patki) [reply supplementary/alica_ros_proxy/CMakeLists.txt:30]: Looks like it had some required imports

File: supplementary/constraintsolver/CMakeLists.txt (hunk 17 above)
Comment (dhananjay-patki) [reply supplementary/constraintsolver/CMakeLists.txt]: Done

File: alica_dummy_proxy/CMakeLists.txt [hunk 18]
```diff
@@ -31,7 +26,7 @@ target_include_directories(alica_dummy_proxy PUBLIC
   $<INSTALL_INTERFACE:include>
 )
 
-target_link_libraries(alica_dummy_proxy alica_engine)
+target_link_libraries(alica_dummy_proxy alica_engine alica_solver_interface)
```
Comment (Dale-Koenig) [inline alica_dummy_proxy/CMakeLists.txt]: Why is this necessary?

**Review by triccyx — APPROVED** [review]
